packages/database/src/exp/Database.ts

```diff
@@ -0,0 +1,143 @@
+import { Reference } from '../api/Reference';
+import { RepoManager } from '../core/RepoManager';
+import { FirebaseAuthInternalName } from '@firebase/auth-interop-types';
+import { Database } from '../api/Database';
+import { Provider } from '@firebase/component';
+
+/**
+ * Class representing a Firebase Realtime Database.
+ */
+export class FirebaseDatabase implements _FirebaseService {
+  static readonly ServerValue = Database.ServerValue;
+
+  private _delegate: Database;
+
+  constructor(
+    readonly app: FirebaseApp,
+    authProvider: Provider<FirebaseAuthInternalName>,
+    databaseUrl?: string
+  ) {
+    this._delegate = RepoManager.getInstance().databaseFromApp(
+      this.app,
+      authProvider,
+      databaseUrl
+    );
+  }
+
+  /**
+   * Modify this instance to communicate with the Realtime Database emulator.
+   *
+   * <p>Note: This method must be called before performing any other operation.
+   *
+   * @param host - the emulator host (ex: localhost)
+   * @param port - the emulator port (ex: 8080)
+   */
+  useEmulator(host: string, port: number): void {
+    this._delegate.useEmulator(host, port);
+  }
+
+  /**
+   * Returns a reference to the root or to the path specified in the provided
+   * argument.
+   *
+   * @param path - The relative string path or an existing Reference to a
+   * database location.
+   * @throws If a Reference is provided, throws if it does not belong to the
+   * same project.
+   * @returns Firebase reference.
+   */
+  ref(path?: string): Reference;
+  ref(path?: Reference): Reference;
+  ref(path?: string | Reference): Reference {
+    return typeof path === 'string'
+      ? this._delegate.ref(path)
+      : this._delegate.ref(path);
+  }
+
+  /**
+   * Returns a reference to the root or the path specified in url.
+   * We throw a exception if the url is not in the same domain as the
+   * current repo.
+   * @param url - A URL that refers to a database location.
+   * @returns A Firebase reference.
+   */
+  refFromURL(url: string): Reference {
+    return this._delegate.refFromURL(url);
+  }
+
+  goOffline(): void {
+    this._delegate.goOffline();
+  }
+
+  goOnline(): void {
+    this._delegate.goOnline();
+  }
+
+  _delete(): Promise<void> {
+    return this._delegate.INTERNAL.delete();
+  }
+
+  _setDatabaseUrl(url: string) {}
+}
+
+const ServerValue = Database.ServerValue;
+export { ServerValue };
+
+/**
+ * Initializes a new instance of the Realtime Database. If a custom database URL
+ * is specified, the URL overrides the database URL provided by the Firebase
+ * App.
+ *
+ * @param app - The {@link FirebaseApp} with which the Realtime Database
+ * instance will be associated with.
+ * @param url - The URL of the Realtime Database instance to connect to. If not
+ * provided, the SDK connects to the default instance of the Firebase App.
+ * @returns A newly initialized `FirebaseDatabase` instance.
+ */
+export function initializeDatabase(
```

Comment:
Do we need it anymore? It's identical to `getDatabase()`

Reply:
I am okay with removing it, but I added it to allow consistency between the different APIs. Ideally, we would throw if `initializeDatabase` gets called multiple times or called after `getDatabase`.

Reply:
`initilizeFoo()` is only needed if you can pass configuration when initializing a SDK, which is not the case for Database. I'd remove it. We can add it back if there is a need in the future. WDYT?

Reply:
Removed for now.